File: 56-merge-intervals/56-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        
    Arrays.sort(intervals, (a1, a2)-> a1[0]-a2[0]);
      //  System.out.println(Arrays.deepToString(intervals));
    
        int st= intervals[0][0];
        int en = intervals[0][1];
        List<int[]> ans =new ArrayList<>();
        for(int i=1;i<intervals.length;i++){
            
            if(en >= intervals[i][0])
                en=Math.max(en, intervals[i][1]);
            else{
                ans.add(new int[]{st,en});
                st= intervals[i][0];
                en= intervals[i][1];
            }
                
        }
    ans.add(new int[]{st,en});
        
        return ans.toArray(new int[ans.size()][2]);
    
    }
}

File: 56-merge-intervals/56-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        
        
        sort(intervals.begin(), intervals.end(),[](const vector<int>&v1,const vector<int>&v2){
           if(v1[0]==v2[0]){
               return v1[1]<v2[1];
           }else{
               return v1[0] < v2[0];
           } 
        });
        
        int st = intervals[0][0];
        int en = intervals[0][1];
        
        vector<vector<int>>ans;
        for(int i=1;i<intervals.size();i++){
            
            int stCurrent = intervals[i][0];
            if(stCurrent <= en ){
                en = max(en, intervals[i][1]);
            }else{
                ans.push_back({st,en});
                st = intervals[i][0];
                en = intervals[i][1]; 
            }
            
        }
          ans.push_back({st,en});
        
        return ans;
        
    }
};